### PR TITLE
fix: TLS enabled leads to zombie tasksd

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
On master restart, having TLS enabled causes task recovery to fail due to a problematic DB scan, leading to tasks still running on an agent without knowledge from the master. This PR changes the data that gets saved so that marshaling and unmarshaling don't break

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/core.go [hunk 1]
```diff
@@ -859,13 +859,14 @@ func (m *Master) Run(ctx context.Context) error {
 
 	cert, err := m.config.Security.TLS.ReadCertificate()
 	if err != nil {
-		return errors.Wrap(err, "failed to read TLS certificate")
+		return err
```
Comment (stoksc) [inline master/internal/core.go:862]: i could see this context being useful. what does it look like if you provide a non-existent TLS cert without it?

**Review by stoksc — APPROVED** [review]
lgtm, one minor comment

File: master/internal/core.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/core.go]: forgot to push some changes, the new ones restore this error message and have some linting

**Comment by eecsliu** [comment]
also thanks @erikwilson for coauthoring

**Pull request merged**